modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -467,27 +470,36 @@ public static void shutdownAndAwaitTermination(ExecutorService service, long tim
      * thrown exception will be propagated to the caller, after all other objects are closed, similar to
      * the try-with-resources block.
      *
-     * @param closeables Collection of objects to close.
+     * @param closeables Stream of objects to close.
      * @throws Exception If failed to close.
      */
-    public static void closeAll(Collection<? extends AutoCloseable> closeables) throws Exception {
-        Exception ex = null;
+    public static void closeAll(Stream<? extends AutoCloseable> closeables) throws Exception {
```

Comment:
Contatenating streams is a lot more convenient then creating new collection or duplicate exception handling logic

Reply:
@ibessonov Just remind you that we don't want to use Stream API in hot paths. I didn't like this method before your changes. Now I hate it :) Just because too many redundant transformations.